SPEC.md

```diff
@@ -0,0 +1,191 @@
+  If this field is not indicated then container runtimes should consider that the JSON file targets all runtimes.
+  If this field is indicated then container runtimes should only run it if one of the identifiers matches the container runtime's identifier.
+  Possible values (not an exhaustive list): docker, podman, gvisor, lxc
+
+#### CDI Devices
+
+The `cdiDevices` field describes the set of hardware devices that can be requested by the container runtime user.
+Note: For a CDI file to be valid, at least one entry must be specified in this array.
+
+  * `cdiDevices` (array of objects, REQUIRED) list of devices provided by the vendor.
+    * `name` (string, REQUIRED), name of the device, can be used to refer to it when requesting a device.
+      * Beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
+      * e.g: `docker/podman run --device foo ...`
+    * `nameShort` (array of strings, OPTIONAL), alternative names for the device. Can be used to reduce the CLI verbosity
+      * Entries in the array MUST use the same schema as the entry for the `name` field
+    * `containerSpec` (object, OPTIONAL) this field is described in the next section.
+      * This field should only be merged in the OCI spec if the device has been requested by the container runtime user.
+
+
+#### Container Spec
+
+The `containerSpec` field describes edits to be made to the OCI specification. Currently only three kinds of edits can be made to the OCI specification: `devices`, `mounts` and `hooks`.
+
+The `containerSpec` field is referenced in two places in the specification:
+  * At the device level, where the edits MUST only be made if the matching device is requested by the container runtime user.
+  * At the container level, where the edits MUST be made if any of the of the device defined in the `devices` field are requested.
+
+
+The `containerSpec` field has the following definition:
+  * `devices` (array of objects, OPTIONAL) describes the device nodes that should be mounted:
+    * `hostPath` (string, REQUIRED) path of the device on the host.
+    * `containerPath` (string, REQUIRED) path of the device within the container.
+    * `permissions` (string, OPTIONAL) Cgroups permissions of the device, candidates are one or more of:
+      * r - allows container to read from the specified device.
+      * w - allows container to write to the specified device.
+      * m - allows container to create device files that do not yet exist.
+  * `mounts` (array of objects, OPTIONAL) describes the mounts that should be mounted:
+    * `hostPath` (string, REQUIRED) path of the device on the host.
+    * `containerPath` (string, REQUIRED) path of the device within the container.
+    * `permissions` (string, OPTIONAL) Cgroups permissions of the device, candidates are one or more of:
```

Comment:
Did you mean to include a list of permissions? personally I don't think this is needed.

Reply:
well you might want to do a readonly mount :) 
Thanks for picking up on the missing text!

Reply:
Updated, based myself on the CRI interface: https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1alpha2/api.proto#L177-L200